packages/trie/src/proof/range.ts

```diff
@@ -382,6 +384,180 @@ async function hasRightElement(trie: Trie, key: Nibbles): Promise<boolean> {
   return false
 }
 
+export function isValueNode(node: TrieNode): boolean {
+  return (
+    node instanceof LeafNode ||
+    (node instanceof BranchNode && node.value() !== null && node.value()!.length > 0)
+  )
+}
+
+export function hasRightBranch(
+  node: BranchNode,
+  nibble: number
+): undefined | [number, EmbeddedNode] {
+  for (let i = nibble + 1; i < 16; i++) {
+    try {
+      const b = node.getBranch(i)
+      if (b !== null) {
+        return [i, b]
+      }
+    } catch (err) {
+      //
+    }
+  }
+  return undefined
```

Comment:
There is syntax somewhere `hasRightBranch() ?? null`, so maybe this should be changed to `return null`.